qiskit/circuit/quantumcircuit.py

```diff
@@ -1997,72 +1997,102 @@ def depth(
         filter_function: Optional[callable] = lambda x: not getattr(
             x.operation, "_directive", False
         ),
+        *,
+        recurse: bool = False,
     ) -> int:
         """Return circuit depth (i.e., length of critical path).
 
         Args:
             filter_function (callable): a function to filter out some instructions.
                 Should take as input a tuple of (Instruction, list(Qubit), list(Clbit)).
                 By default filters out "directives", such as barrier or snapshot.
+            recurse: if ``True``, then recurse into control-flow operations.  For loops
+                with known-length iterators are counted unrolled.  If-else blocks take the
+                longer case of the two branches.  While loops are counted as if the condition runs
```

Comment:
```suggestion
                longer case of the two branches.  While loops are counted as if the loop body runs
```

Reply:
We decided to remove the public `QuantumCircuit` changes from this release (`DAGCircuit` generally has less public visibility, so we're not as worried by that).